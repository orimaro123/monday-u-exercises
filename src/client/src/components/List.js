import React, { useEffect, useState } from "react";
import ListControls from "./ListControls";

import { fetchAllItems } from "../itemClient";
import Item from "./Item";

const List = ({ allItems , itemToDelete , itemToEdit}) => {
  
  return (
		<div>
			{allItems.map((item) => {
				return (
					<Item
						key={item.itemId}
						item={item}
            itemToDelete={itemToDelete}
            itemToEdit={itemToEdit}
				
					/>
				);
			})}
		</div>
	);

};
export default List;
