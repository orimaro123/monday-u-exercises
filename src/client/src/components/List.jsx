import React from "react";
import PropTypes from "prop-types";
import Item from "./Item";

const List = ({ items }) => {
  return (
    <div>
      {items.map((item) => {
        return (
          <Item
            key={item.itemId}
            item={item}
            //itemToDelete={itemToDelete}
            //itemToEdit={itemToEdit}
          />
        );
      })}
    </div>
  );
};

export default List;
