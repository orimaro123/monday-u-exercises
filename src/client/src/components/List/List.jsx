import React from "react";
import ItemConnector from "../Item/ItemConnector";
import useFilterByQuery from "../../hooks/useFilterByQuery";

const List = ({ items }) => {
  const filter = useFilterByQuery();

  return (
    <ul>
      {filter(items).map((item) => {
        return <ItemConnector key={item.id} item={item} />;
      })}
    </ul>
  );
};

export default List;
