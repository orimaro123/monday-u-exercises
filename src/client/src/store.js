import { configureStore } from "@reduxjs/toolkit";
import thunkMiddleware from "redux-thunk";
import allReducers from "./redux/reducers";

const logger = store => next => action => {
  console.log('dispatching', action)
  
  let result = next(action)
  console.log('next state', store.getState())
  return result
}

export const store = configureStore({
  reducer: allReducers,
  middleware: [thunkMiddleware, /* logger */],
  preloadedState: {}
});