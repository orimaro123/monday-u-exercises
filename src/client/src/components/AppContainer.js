import React, { useEffect, useState } from "react";
import ListControls from "./ListControls";
import List from "./List";

import {
  fetchAllItems,
  createItem,
  deleteItemById,
  updateNameInDb,
  clearAll,
} from "../itemClient";
import ComponentItemsRender from "./List";

const AppContainer = () => {
  const [allItems, setAllItems] = useState([]);
  const [hideClass, setHideClass] = useState("hide");
  const [clearAllHideClass, setClearAllHideClass] = useState("hide");

  const itemToCreate = async (item) => {
    setHideClass("");
    await createItem(item);
    const items = await fetchAllItems();
    setAllItems(items.data);
    setHideClass("hide");
    setClearAllHideClass("");
  };

  const itemToDelete = async (itemId) => {
    setHideClass("");
    await deleteItemById(itemId);
    const items = await fetchAllItems();
    setAllItems(items.data);
    setHideClass("hide");

    if (items.data.length === 0) {
      setClearAllHideClass("hide");
    }
  };

  const itemToEdit = async (itemId, newName) => {
    setHideClass("");
   let itemNewName = await updateNameInDb(itemId, newName);
  
    const items = await fetchAllItems();
    setAllItems(items.data);
    setHideClass("hide");
   
 
  return itemNewName
   
  };

  const clearAllItems = async () => {
    setAllItems([]);
    await clearAll();
    setClearAllHideClass("hide");
  };

  useEffect(() => {
    const fetchedItems = async () => {
      setHideClass("");
      const items = await fetchAllItems();

      setAllItems(items.data);
      if(items.data.length > 0){
    setClearAllHideClass("");

      }

      setHideClass("hide");
    };
    fetchedItems();
  }, []);

  return (
    <div className="app-container">
      <div className="list-container-background">
        <div className="app-name">Ori's Todo List</div>

        <ListControls itemToCreate={itemToCreate} />
        <div className="list-container">
          <List
            itemToEdit={itemToEdit}
            itemToDelete={itemToDelete}
            allItems={allItems}
          />
        </div>

        <div className={`loader-wrapper ${hideClass}`}>
          <span className="loader  ">
            <span className="loader-inner "></span>
          </span>
        </div>
        <div className="clear-all-button-div">
          <button
            onClick={() => clearAllItems()}
            className={`clear-all-button ${clearAllHideClass}`}
          >
            ClearAll
          </button>
        </div>
      </div>
    </div>
  );
};
export default AppContainer;
