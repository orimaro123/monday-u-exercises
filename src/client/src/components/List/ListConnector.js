import { bindActionCreators } from "redux";
import { connect } from "react-redux";

import List from "./List";
import { getItems } from "../../redux/selectors/itemsEntitiesSelectors";

const mapStateToProps = (state, ownProps) => {
  const items = getItems(state);

  return { items };
};


const mapDispatchToProps = (dispatch, ownProps) => {
  return bindActionCreators(
    {

    },
    dispatch
  );
};

export default connect(mapStateToProps, mapDispatchToProps)(List);
