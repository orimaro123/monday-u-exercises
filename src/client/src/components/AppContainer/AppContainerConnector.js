import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import {
  getShowClearButton,
  getShowToast,
  getToastContent,
  getToastOrientation,
} from "../../redux/selectors/itemsViewSelectors";

import AppContainer from "./AppContainer";
import { hideToastAction } from "../../redux/actions/itemsViewsActions";

import {
  getItemsAction,
  clearAllItemsAction,
} from "../../redux/actions/itemsEntitiesActions";

const mapStateToProps = (state, ownProps) => {
  const showClearButton = getShowClearButton(state);
  const showToast = getShowToast(state);
  const toastContent = getToastContent(state);
  const toastOrientation = getToastOrientation(state);

  return {
    showClearButton,
    showToast,
    toastContent,
    toastOrientation,
  };
};

const mapDispatchToProps = (dispatch, ownProps) => {
  return bindActionCreators(
    {
      hideToastAction,

      getItemsAction,
      clearAllItemsAction,
    },
    dispatch
  );
};

export default connect(mapStateToProps, mapDispatchToProps)(AppContainer);
