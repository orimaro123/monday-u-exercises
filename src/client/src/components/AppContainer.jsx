import React, { useEffect, useState, useCallback } from "react";
import ListControls from "./ListControls";
import ListControlsConnector from "./ListControlsConnector";
import List from "./List";
import { Button, Toast } from "monday-ui-react-core";

import {
  fetchAllItems,
  createItem,
  deleteItemById,
  updateNameInDb,
  clearAll,
} from "../services/itemClient";
import Search from "./Search";
import ListConnector from "./ListConnector";

const AppContainer = ({
  showLoaderAction,
  hideLoaderAction,
  showLoader,
  showClearButtonAction,
  hideClearButtonAction,
  showClearButton,
  showToastAction,
  hideToastAction,
  showToast,
  crateItemAction,
  getItemsAction,
  clearAllItemsAction,
  toastContent,
 
}) => {
  const [allItems, setAllItems] = useState([]);


  useEffect(() => {
    getItemsAction();
  }, [getItemsAction]);

  const onCloseCallback = useCallback(() => {
    hideToastAction();
  }, [hideToastAction]);

  return (
    <div className="app-container">
      <div className="upper-div">
        <Search />
      </div>

      <div className="list-container-background">
        <div className="app-name">Ori's List</div>

        <ListControlsConnector />

        <div className="list-container">
          <ListConnector
          //itemToEdit={itemToEdit}
          />
        </div>

        <div className="clear-all-button-div">
          {showClearButton ? (
            <Button
              onClick={clearAllItemsAction}
              id="clear-all-button"
              className={`clear-all-button `}
            >
              Clear All
            </Button>
          ) : null}
          <Toast
            open={showToast}
            type={Toast.types.POSITIVE}
            onClose={() => onCloseCallback()}
            autoHideDuration={4000}
            className="monday-storybook-toast_box"
          >
            {toastContent}
          </Toast>
        </div>
      </div>
    </div>
  );
};
export default AppContainer;
