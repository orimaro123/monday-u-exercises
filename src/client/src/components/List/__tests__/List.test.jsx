import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { Provider } from "react-redux";
import { store } from "../../../store";
import List from "../List";
import AppContainer from "../../AppContainer/AppContainer";

const items = [
  {
    id: 56,
    itemName: "Take dog out for a walk",
    text: "Take dog out for a walk",
    status: false,
    doneAt: "2022-07-13 07:31:33",
  },
  {
    id: 32,
    itemName: "Do the dishes",
    text: "Do the dishes",
    status: true,
    doneAt: "2022-07-13 09:31:33",
  },
];

describe("List", () => {
  test("should render both items (one done and one not)", () => {
    render(
      <Provider store={store}>
        <List items={items} fetchItems={jest.fn(() => items)} />
      </Provider>
    );

    const item1 = screen.getByTestId(`item-${items[0].id}`);
    const item2 = screen.getByTestId(`item-${items[1].id}`);

    expect(item1).toBeInTheDocument();
    expect(item1).toMatchSnapshot();
    expect(item2).toBeInTheDocument();
    expect(item2).toMatchSnapshot();
  });

  test("should call fetchItems function", () => {
    const getItemsAction = jest.fn(() => items);
    render(
      <Provider store={store}>
        <AppContainer items={items} getItemsAction={getItemsAction} />
      </Provider>
    );
    expect(getItemsAction).toHaveBeenCalled();
  });
});
