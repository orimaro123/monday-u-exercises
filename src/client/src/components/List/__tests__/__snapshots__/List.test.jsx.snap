// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render both items (one done and one not) 1`] = `
<input
  class="list-item-text "
  data-testid="item-56"
  readonly=""
  type="text"
  value="Take dog out for a walk"
/>
`;

exports[`List should render both items (one done and one not) 2`] = `
<input
  class="list-item-text decorate"
  data-testid="item-32"
  readonly=""
  type="text"
  value="Do the dishes"
/>
`;
